pkg/model/model.go

```diff
@@ -4,10 +4,16 @@ package model
 
 // Resource - Describes a resource (node)
 type Resource struct {
-	Kind           string `json:"kind,omitempty"`
-	UID            string `json:"uid,omitempty"`
-	ResourceString string `json:"resourceString,omitempty"`
-	Properties     map[string]interface{}
+	UID            string          `json:"uid,omitempty"`
+	ResourceString string          `json:"resourceString,omitempty"`
+	Properties     PropertiesArray `json:"properties"`
```

Comment:
I don't think we can change Properties from `map[string]interface{}`.  The object won't have a way to access the other properties.

Reply:
Can't we use the `PropertiesArray` to access all other properties?

Reply:
An alternative is to add a new field, like `PrimaryProperties`.

Reply:
> Can't we use the PropertiesArray to access all other properties?
`PropertiesArray` only declares 3 properties, it doesn't have a way to access the other properties. We also don't want to declare each individual property because these are different for each kind.

Reply:
Got it. We need to keep Properties as a map within Resources struct and then make new struct for the specific Property fields.